src/Symfony/Component/VarDumper/Caster/ReflectionCaster.php

```diff
@@ -76,7 +76,20 @@ public static function castClosure(\Closure $c, array $a, Stub $stub, $isNested)
 
     public static function castGenerator(\Generator $c, array $a, Stub $stub, $isNested)
     {
-        return class_exists('ReflectionGenerator', false) ? self::castReflectionGenerator(new \ReflectionGenerator($c), $a, $stub, $isNested) : $a;
+        if (!class_exists('ReflectionGenerator', false)) {
+            return $a;
+        }
+
+        // Cannot create ReflectionGenerator based on a terminated Generator
+        try {
+            $reflectionGenerator = new \ReflectionGenerator($c);
+        } catch (\Exception $e) {
```

Comment:
I confirm that an `Exception` is thrown here, even-though I'd personally have expected a `ReflectionException`
ping @jpauli.